### PR TITLE
Convert code to PEP8 and add lint/syntax check

### PR DESCRIPTION
**Pull request by mattbostock**
Add a `.travis.yml` file to have Travis CI run flake8 whenever a pull
request is raised.

Travis will run `make test`, which in turn will run `flake8`, which
wraps the PyFlakes syntax checker and pep8 style checker:
https://pypi.python.org/pypi/flake8

We test `fabfile.py` separately to all other files because it imports
the Fabfile subcommand modules, which causes flakes8 to report:

    fabfile.py:16:1: F401 'puppet' imported but unused
    fabfile.py:19:1: F401 'app' imported but unused
    fabfile.py:20:1: F401 'apt' imported but unused
    fabfile.py:21:1: F401 'cache' imported but unused
    fabfile.py:22:1: F401 'campaigns' imported but unused
    fabfile.py:23:1: F401 'cdn' imported but unused
    fabfile.py:24:1: F401 'elasticsearch' imported but unused
    fabfile.py:25:1: F401 'incident' imported but unused
    fabfile.py:26:1: F401 'licensify' imported but unused
    fabfile.py:27:1: F401 'locksmith' imported but unused
    fabfile.py:28:1: F401 'logstream' imported but unused
    fabfile.py:29:1: F401 'mongo' imported but unused
    fabfile.py:30:1: F401 'mainstream_slugs' imported but unused
    fabfile.py:31:1: F401 'mapit' imported but unused
    fabfile.py:32:1: F401 'mysql' imported but unused
    fabfile.py:33:1: F401 'nagios' imported but unused
    fabfile.py:34:1: F401 'nginx' imported but unused
    fabfile.py:35:1: F401 'ntp' imported but unused
    fabfile.py:36:1: F401 'performanceplatform' imported but unused
    fabfile.py:37:1: F401 'postgresql' imported but unused
    fabfile.py:38:1: F401 'rabbitmq' imported but unused
    fabfile.py:39:1: F401 'rkhunter' imported but unused
    fabfile.py:40:1: F401 'search' imported but unused
    fabfile.py:41:1: F401 'statsd' imported but unused
    fabfile.py:42:1: F401 'topic_change' imported but unused
    fabfile.py:43:1: F401 'vm' imported but unused
    fabfile.py:44:1: F401 'whitehall' imported but unused

So we run flake8 separately against `fabfile.py`, ignoring errors of
type `F401` ("imported but not used").

For all other files, we ignore:

  - `E501` "line too long" errors, since modern terminals are capable of
    displaying very long lines and enforcing shorter lines using tests
    can hamper readability

  - `F403` "'from module import *' used; unable to detect undefined
    names", which commonly occurs in our code but seems to be spurious;
    I'm not familiar enough with Python to judge

* * *

__I'm not very familiar with Python or autopep8 and have only performed minimal testing on these changes; I would appreciate feedback from someone with more experience with autopep8.__

### CODE REVIEW

**Comment by mattbostock** [comment]
Spoke to @danielroseman about this IRL; I'm going to re-do the autopep8 commits to make them change just enough code to make the tests pass. Some of the changes appear to make the readability worse, so I'm going to try to avoid those changes by passing options to autopep8.

**Comment by mattbostock** [comment]
I've rebased to greatly reduce the number of changes and ignore PEP8 errors which might hamper readability.

File: performanceplatform.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def get_command(query_path, query):
     pp_collector = './venv/bin/pp-collector'
     credentials_path = './config/credentials/{}.json'.format(credentials)
     token_path = './config/tokens/{}.json'.format(
-    query['token'])
+        query['token'])
```
Comment (danielroseman) [inline performanceplatform.py]: this could be brought up to the previous line.

File: topic_change.py [hunk 2]
```diff
@@ -173,7 +175,7 @@ def prepare_whitehall_csv(output_directory, source_topic, destination_topic):
               SOURCE=source_topic,
               DESTINATION=destination_topic,
               CSV_LOCATION=remote_path,
-    )
+              )
```
Comment (danielroseman) [inline topic_change.py]: I prefer the original here. Was it causing a flake failure?

**Comment by danielroseman** [comment]
Apart from a couple of very minor comments, :+1:

**Comment by bradwright** [comment]
I'm not sure we should ignore `F403` - it's a smell. It'd be better to be explicit about which modules we're adding to the namespace.

File: topic_change.py (hunk 2 above)
Comment (mattbostock) [inline topic_change.py]: Yes: `E124 closing bracket does not match visual indentation`

For what it's worth, vim also agrees with autopep8 as to the indentation. That's not to say we can't change it, but we'd need to tell flake to ignore it. I think it's better to leave as-is.

File: performanceplatform.py (hunk 1 above)
Comment (mattbostock) [inline performanceplatform.py]: Fixed in f30392c.

**Comment by mattbostock** [comment]
@bradwright: I've added a `FIXME` to address `F403` errors in fd8a3fc. I'd rather not let those errors block this pull request.

**Comment by bradwright** [comment]
:+1: from me

**Comment by mattbostock** [comment]
@danielroseman Are you happy to merge?

**Pull request merged**